File: 242-valid-anagram/valid-anagram.ts
function isAnagram(s: string, t: string): boolean {
    if(s.length !== t.length) return false;
    const map = {};
    for(let i = 0; i < s.length; i++) {
        if(map[s[i]] !== undefined) map[s[i]] ++
        else map[s[i]] = 1;

        if(map[t[i]] !== undefined) map[t[i]] --
        else map[t[i]] = -1;
    }
    return Object.values(map).every(value=>value === 0)
};

File: 242-valid-anagram/valid-anagram.go
func isAnagram(s string, t string) bool {
    if len(s) != len(t) {
        return false;
    }
    counter := make([]int, 26)

    for i, val := range s {
        counter[val - 97]++;
        counter[t[i] - 97]--;
    }

    for _, val := range counter {
        if val != 0 {
            return false;
        }
    }
    return true;
}